### PR TITLE
Add `GAP_NewRange`, `GAP_NewObjIntFromInt`, `GAP_ValueInt` to the libgap API

### PR DESCRIPTION
**Pull request by fingolfin**
This is cleanup, but it will also be used in the GAP-Julia interface. Might also be useful for other libgap users.

<s>TODO: This needs some more work, in particular the libgap API functions needs to be documented.</s>

### CODE REVIEW

File: src/libgap-api.c [hunk 1]
```diff
@@ -406,6 +407,21 @@ Obj GAP_NewPlist(Int capacity)
...
+static BOOL fitsInIntObj(Int i)
+{
+    return INT_INTOBJ_MIN <= i && i <= INT_INTOBJ_MAX;
+}
+
+Obj GAP_NewRange(Int len, Int low, Int inc)
+{
+    if (!fitsInIntObj(len)) return 0;
+    if (!fitsInIntObj(low)) return 0;
+    if (!fitsInIntObj(inc)) return 0;
```
Comment (ChrisJefferson) [inline src/libgap-api.c:419]: Do you just want to catch inc==0 here too? I don't think it's allowed.

Comment (fingolfin) [reply src/libgap-api.c]: Good point, done

**Comment by fingolfin** [comment]
I've now enhanced this a bit more to resolves issue #4208 by @embray who also may have comments on this PR.

**Review by ThomasBreuer — APPROVED** [review]
Just for curiosity: Why `fitsInIntObj` and not `fitsInObjInt`?
(`ObjInt` is used in most situations in the `src` files.)

(And there are two instances of "an GAP integer object" in comments.)

**Comment by fingolfin** [comment]
@ThomasBreuer "intobj" refers to immediate ints (cf. `IS_INTOBJ`, `INT_INTOBJ`, and many more), while objint refers to any kind of GAP integer, i.e., "an `Obj` representing an integer"; the latter is relatively rarely used, though, mostly in `ObjInt_Int` and `Int_ObjInt` (and there many variants; but the variants all derive from those two; as does `MakeObjInt`).

It is rather confusing.

**Pull request merged**